airflow/models/skipmixin.py

```diff
@@ -218,10 +226,16 @@ def skip_all_except(
             for branch_task_id in list(branch_task_id_set):
                 branch_task_id_set.update(dag.get_task(branch_task_id).get_flat_relative_ids(upstream=False))
 
-            skip_tasks = [t for t in downstream_tasks if t.task_id not in branch_task_id_set]
-            follow_task_ids = [t.task_id for t in downstream_tasks if t.task_id in branch_task_id_set]
+            skip_tasks = []
+            for t in downstream_tasks:
+                downstream_ti = dag_run.get_task_instance(  # type: ignore[union-attr]
+                    t.task_id, map_index=ti.map_index
+                )
+                if downstream_ti and t.task_id not in branch_task_id_set:
+                    skip_tasks.append((t.task_id, downstream_ti.map_index))
```

Comment:
This can probably be rewritten into a list comprehension

Reply:
It will be something like that, is it ok?
```
skip_tasks = [
                (t.task_id, downstream_ti.map_index)
                for t in downstream_tasks
                if (downstream_ti := dag_run.get_task_instance(t.task_id, map_index=ti.map_index))
                and t.task_id not in branch_task_id_set
            ]
```